pkg/monitor/datapath_trace.go

```diff
@@ -108,16 +111,25 @@ func (n *TraceNotify) traceSummary() string {
 	}
 }
 
+// OriginalIP returns the original source IP if reverse NAT was performed on
+// the flow
+func (n *TraceNotify) OriginalIP() net.IP {
+	if n.Flags != 0 {
```

Comment:
Please check if `if n.Flags & ipv6Flag != 0` instead to avoid long debugging sessions in the future if we add one more flag.